src/lib/core/components/visualizations/InputVariables.tsx

```diff
@@ -82,28 +83,60 @@ const useStyles = makeStyles(
 );
 
 export function InputVariables(props: Props) {
-  const { inputs, entities, values, onChange } = props;
+  const { inputs, entities, values, onChange, constraints } = props;
   const classes = useStyles();
-  const handleChange = (inputName: string, value?: VariableDescriptor) => {
+  const handleChange = (inputName: string, value?: Variable) => {
     onChange({ ...values, [inputName]: value });
   };
+  const flattenedConstraints =
+    constraints && flattenConstraints(values, entities, constraints);
+
+  // Steps
```

Comment:
I don't really get what this is about, sorry!
It's something to do with providing the entity/variable data tree to each menu in the InputVariables interface. Is the ordering something to do with `dataElementDependencyOrder`? It can't be, because this is unused at the moment.

Reply:
This does have to do with `dataElementDependencyOrder`. Line 99 should begin with
```typescript
   const rootEntities = dataElementDependencyOrder.map((inputName, index) => {
```

Good catch!

Reply:
I will work on fixing this up

Reply:
I pushed a fix for this. Not exactly what I put above.

Reply:
Glad to have helped, without fully understanding this part!